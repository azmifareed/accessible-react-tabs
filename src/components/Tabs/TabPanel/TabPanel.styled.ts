import styled from 'styled-components';

export interface IContainer {
  isSelected: boolean;
}

export const Container = styled.div<IContainer>`
  display: ${p => p.isSelected ? 'flex' : 'none'};
  flex-direction: column;
  padding: ${p => p.theme.spacing.large};
`