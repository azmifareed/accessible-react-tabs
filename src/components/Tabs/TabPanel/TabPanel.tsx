import React, { ReactNode } from 'react';
import { Container } from './TabPanel.styled';

export interface ITabPanelProps {
  id: string;
  tabIndex?: number;
  ariaLabelledby: string;
  isSelected?: boolean;
  children?: ReactNode;
}

export function TabPanel ({ id, tabIndex, ariaLabelledby, isSelected, children }: ITabPanelProps) {
  return (
    <Container role="tabpanel" tabIndex={tabIndex} id={id} aria-labelledby={ariaLabelledby} isSelected={!!isSelected}>
      {children}
    </Container>
  );
}
