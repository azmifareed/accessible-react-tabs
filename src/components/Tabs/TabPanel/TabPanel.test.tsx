import React from 'react';
import { render } from '../../../utils/test-utils';
import { TabPanel } from './TabPanel';

test('renders TabPanel correctly', () => {
  const component = render(<TabPanel 
    id="test-tab" 
    tabIndex={0} 
    ariaLabelledby="test" 
    isSelected={true}>test child</TabPanel>)
  expect(component.baseElement).toMatchSnapshot();
});

test('renders TabPanel correctly when not selected', () => {
  const component = render(<TabPanel 
    id="test-tab" 
    tabIndex={0} 
    ariaLabelledby="test" 
    isSelected={false}>test child</TabPanel>)
  expect(component.baseElement).toMatchSnapshot();
});


