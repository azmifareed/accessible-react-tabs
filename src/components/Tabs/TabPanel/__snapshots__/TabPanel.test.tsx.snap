// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TabPanel correctly 1`] = `
<body>
  <div>
    <div
      aria-labelledby="test"
      class="sc-bdnxRM klgMsN"
      id="test-tab"
      role="tabpanel"
      tabindex="0"
    >
      test child
    </div>
  </div>
</body>
`;

exports[`renders TabPanel correctly when not selected 1`] = `
<body>
  <div>
    <div
      aria-labelledby="test"
      class="sc-bdnxRM jYqWsk"
      id="test-tab"
      role="tabpanel"
      tabindex="0"
    >
      test child
    </div>
  </div>
</body>
`;
